setup.go

```diff
@@ -118,22 +139,124 @@ func Setup(ctx context.Context, cfg *Config) (*Node, error) {
 
 	bsctx := metri.CtxScope(ctx, "rainbow")
 
-	r1, err := routingv1client.New(cfg.RoutingV1, routingv1client.WithStreamResultsRequired())
-	if err != nil {
-		return nil, err
-	}
-
-	vs := httpcontentrouter.NewContentRoutingClient(r1)
+	var pr routing.PeerRouting
+	var vs routing.ValueStore
+	var cr routing.ContentRouting
 
 	opts = append(opts, libp2p.Routing(func(h host.Host) (routing.PeerRouting, error) {
-		return vs, nil
+		if cfg.RoutingV1 != "" {
+			routingClient, err := delegatedHTTPContentRouter(cfg.RoutingV1, routingv1client.WithStreamResultsRequired())
+			if err != nil {
+				return nil, err
+			}
+			pr = routingClient
+			vs = routingClient
+			cr = routingClient
+		} else {
+			// If there are no delegated routing endpoints run an accelerated Amino DHT client and send IPNI requests to cid.contact
+
+			// TODO: This datastore shouldn't end up containing anything anyway so this could potentially just be a null datastore
```

Comment:
I don't understand this... aren't dht records stored in this datastore?

Reply:
I mean, I thought that anything retrieved from the DHT would be stored locally too, even if you are only a client.

Reply:
We can double check, but there's no reason to implement what is effectively record caching at this layer. That's not to say the code won't do that (I suspect at least for IPNS records it will) just that it's mostly non-sensical (e.g. IPNS records caching is better done at a layer that understands IPNS records like namesys)